### ISSUE TITLE
Yul compilation failed: * Line 3, Column 42\n  Bad escape sequence in string\nSee Line 3, Column 16643 for detail.\n* Line 5, Column 38\n  Extra non-whitespace after JSON value.\n

### ISSUE THREAD

**Issue by agroce** [ethereum/fe#252]
Compiling this file:

```
contract Foo:
    event MyEvent:
        s: string26
        u: u256
        s: string42
        s: string100
        a: address
        s: string13
        s: string100
    pub def __init__(s1: string42, s2: string26, u: u256,
                     s3:string100, a:address):
      emit MyEvent(s2,u,s1,s3,a,"stat\c string",string100(""))
    pub def return_casted_static_string() -> string100:
      return string100("foo")
```

```
thread 'main' panicked at 'Yul compilation failed: {"errors":[{"component":"general","formattedMessage":"* Line 3, Column 42\n  Bad escape sequence in string\nSee Line 3, Column 16643 for detail.\n* Line 5, Column 38\n  Extra non-whitespace after JSON value.\n","message":"* Line 3, Column 42\n  Bad escape sequence in string\nSee Line 3, Column 16643 for detail.\n* Line 5, Column 38\n  Extra non-whitespace after JSON value.\n","severity":"error","type":"JSONError"}]}', compiler/src/lib.rs:49:27
stack backtrace:
   0: rust_begin_unwind
             at /rustc/c0b64d97beebb09325b5587abed39f4f1621026f/library/std/src/panicking.rs:493:5
   1: std::panicking::begin_panic_fmt
             at /rustc/c0b64d97beebb09325b5587abed39f4f1621026f/library/std/src/panicking.rs:435:5
   2: fe_compiler::compile
   3: fe::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Aborted
```

On latest github master, built with

```
[profile.dev]
debug = true
panic = "abort"
```

for fuzzing.  Using https://github.com/agroce/afl-compiler-fuzzer + afl.rs.

**Comment by cburgdorf** [comment]
This is a case of #329

**Comment by sbillig** [comment]
This is fixed by the new parser (#346). Thanks @agroce!

```
error: String contains an invalid escape sequence
   ┌─ ../eth/badescape.fe:12:33
   │
12 │       emit MyEvent(s2,u,s1,s3,a,"stat\c string",string100(""))
   │                                 ^^^^^^^^^^^^^^^
```
